api/object.rb

```diff
@@ -27,7 +27,13 @@ module Properties
 
       include Properties
 
+      # original value of :name before the provider override happens
+      # same as :name if not overridden in provider
+      attr_reader :api_name
```

Comment:
@nelsonjr I know you also have the concept of `field` for properties. However, `field` is optional and push the logic of which name to use down in the templates. I prefer this approach. Happy to rename or discuss this idea.

That would also be nice if `api_name` could not be set in the yaml file. Is that possible?

Reply:
Possible. We could add a pre_validate method and call it in provider/config (self.parse)

Reply:
I will add a TODO for now. I don't think this is very high priority. Thanks for the idea.